src/Engine/GameAction.java

```diff
@@ -976,4 +976,56 @@ public UnitActionType getUnitActionType()
       return type;
     }
   } // ~TransformAction
+
+  // ===========  UnitDeleteAction  =================================
+  /** Removes the unit. Only allows deletion in place */
+  public static class UnitDeleteAction implements GameAction
+  {
+    final Unit actor;
+    final XYCoord destination;
+    
+    public UnitDeleteAction(Unit unit)
+    {
+      actor = unit;
+      destination = new XYCoord(unit.x, unit.y);
+    }
+
+    @Override
+    public GameEventQueue getEvents(MapMaster gameMap)
+    {
+      GameEventQueue eventSequence = new GameEventQueue();
+      eventSequence.add(new UnitDieEvent(actor));
+      return eventSequence;
+    }
+
+    @Override
+    public ActionType getType()
+    {
+      return ActionType.UNIT;
+    }
+    
+    @Override
+    public String toString()
+    {
+      return String.format("[Delete %s in place]", actor.toStringWithLocation());
```

Comment:
Why "in place"?

Reply:
Because I require the unit to stand still to delete. And because it was copy/pasted from text with a location in it.